how-it-works/data-and-provider-model/buckets-files-and-merkle-proofs.md

```diff
@@ -0,0 +1,108 @@
+
+!!! note
+    Updating file bytes yields a new Merkle root (a new commitment). Renames/metadata updates that don’t touch content don’t change the root.
+
+## Merkle Proofs: Verifying Without Trust
+
+A Merkle proof is a short witness that a specific chunk (or an entire file) is part of a dataset committed by a known root. Given the leaf hash (chunk), its sibling path (with positions), and the on-chain root, anyone can recompute the root and check it matches—no trusted third party needed. Proof size scales as O(log N) in the number of chunks.
+
+- **Challenge leaf size (protocol):** For storage validation, the network uses a protocol-defined challenge chunk size (currently 1 KB).
+- **Client I/O chunk size (implementation):** Upload/download defaults are 8 KB and may be tuned without affecting challenge logic.
+
+**Minimal proof shape (conceptual):**
+```json
+{
+  "fileRoot": "0xROOT...",
+  "leafIndex": 1337,
+  "leafHash": "0xLEAF...",
+  "path": ["0xSIBLING1...", "0xSIBLING2...", "..."],
+  "positions": ["R","L","L","..."],
+  "hashFn": "keccak256" // protocol-defined; value shown for clarity
+}
+```
+
+A verifier recomputes the root from `leafHash`, `leafIndex`, `path` + `positions`; if it equals `fileRoot` (anchored on-chain), the proof is valid.
+
+In DataHaven, Merkle proofs support two essential checks:
+
+  - Integrity at retrieval (MSP read path): When a client downloads a file/chunk from the MSP, the provider includes a proof so the client can verify against the known file root.
+
+  - Ongoing storage proofs (BSP audits): The protocol challenges BSPs on random chunk indices; BSPs answer with Merkle proofs to demonstrate continued custody.
+
+## What’s On-Chain vs Off-Chain
+
+- **Bucket state (on-chain, per bucket via MSP):** A compact cryptographic map (Merkle/Patricia-style) whose root is stored on-chain. MSP updates this root during storage request acceptance and file-deletion flows, supplying proofs for correctness in the same session as data changes.
+
+- **File commitments (on-chain reference):** Each file is identified by its Merkle root and included in the bucket map.
+
+- **BSP commitment (on-chain, per BSP)**: A single global root that summarizes all files a BSP stores (used for randomized challenges and slashing logic).
+
+- **Data (off-chain):** Stored and served by the MSP; replicated by BSPs. Verifiers don’t need the bytes- proofs are enough.
+
+## Read / Write / Migrate Flow
+
+- **Write:** Client → MSP; MSP forwards to BSPs; MSP updates bucket root on-chain with proofs. If a replica declines or anchoring fails, the write aborts.
+
+- **Read:** Client fetches from the MSP and verifies Merkle proofs locally. BSPs are not in the normal read path.
+
+- **Migrate**: If an MSP fails, the user can reassign the bucket to a new MSP. The new MSP can reconstruct from BSP replicas.
+
+## User Personas
```

Comment:
I have the same problem with this title as the previous one. What is this section describing?

Reply:
agreed I'm going to nuke this last section